RFD.md

```diff
@@ -0,0 +1,191 @@
+---
+authors: David Thomason (dtom90@gmail.com)
+state: draft
+---
+
+# RFD - Remote Linux Process Manager
+
+## Required Approvers
+* Engineering: @smallinsky @gabrielcorado @russjones @zmb3 @rosstimothy @r0mant
+
+## What
+
+Implement a prototype job worker service that provides an API to run arbitrary Linux processes.
+
+## Why
+
+Teleport will use this project in order to evaluate the author's eligibility for the Senior Systems Engineer role. Implementing this application simulating a real-world project will allow the author to effectively demonstrate his skills to Teleport.
+
+## Scope
+
+The following will be implemented in this project:
+
+- A Worker library for interacting with jobs
+  - resource control for CPU, Memory and Disk IO per job using cgroups
+- An API server that wraps the functionality of the library
+  - gRPC API
+  - mTLS authentication
+  - simple authorization scheme
+- A command line interface (CLI) that communicates with the API server
+  - start/stop/query status and get the streaming output of a job
+
+The following could be implemented in a real production application, but are out of scope for this project:
+
+- Persistent storage
+  - (storage will be in memory)
+- Segmented user process spaces
+  - (all users will see the same processes)
+
+## Details
+
+### CLI
+
+Users will be able to run a CLI program that will manage jobs on a remote Linux system. This CLI tool will be called `rps` and will have the following commands:
+
+- `rps start -h <host> -c "<command>"`: Start a new job with the given command and return the job_id
```

Comment:
You don't really need `-h <host>` you can hardcode this.

Reply:
dropped -h flag, hard-coding instead